src/ui/public/agg_types/metrics/lib/parent_pipeline_agg_helper.js

```diff
@@ -0,0 +1,61 @@
+import metricAggTemplate from 'ui/agg_types/controls/sub_agg.html';
+import _ from 'lodash';
+import VisAggConfigProvider from 'ui/vis/agg_config';
+import VisSchemasProvider from 'ui/vis/schemas';
+import { parentPipelineAggController } from './lib/parent_pipeline_agg_controller';
+import { parentPipelineAggWritter } from './lib/parent_pipeline_agg_writter';
+
+const ParentPipelineAggHelperProvider = function (Private) {
+
+  const AggConfig = Private(VisAggConfigProvider);
+  const Schemas = Private(VisSchemasProvider);
+
+  const metricAggFilter = ['!top_hits', '!percentiles', '!percentile_ranks', '!median', '!std_dev'];
+  const metricAggSchema = (new Schemas([
+    {
+      group: 'none',
+      name: 'metricAgg',
+      title: 'Metric Agg',
```

Comment:
right now its only used by derivative (and comulative sum), we'll be adding other pipeline aggregations shortly and they all use this.

its the metric you define on the derivative (either custom metric or one of the pre existing metric aggs) .... list is filtered down as not all metrics are supported with derivative (you can't calculate a derivative on top hits for example)